app/code/Magento/ConfigurableProduct/Plugin/Model/ResourceModel/Product.php

```diff
@@ -129,18 +129,17 @@ private function resetConfigurableOptionsData($object)
      * Gather configurable parent ids of product being deleted and reindex after delete is complete.
      *
      * @param \Magento\Catalog\Model\ResourceModel\Product $subject
-     * @param \Closure $proceed
+     * @param \Magento\Catalog\Model\ResourceModel\Product $result
      * @param \Magento\Catalog\Model\Product $product
      * @return \Magento\Catalog\Model\ResourceModel\Product
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function aroundDelete(
+    public function afterDelete(
         \Magento\Catalog\Model\ResourceModel\Product $subject,
-        \Closure $proceed,
+        \Magento\Catalog\Model\ResourceModel\Product $result,
         \Magento\Catalog\Model\Product $product
     ) {
         $configurableProductIds = $this->configurable->getParentIdsByChild($product->getId());
```

Comment:
not sure this still will be available after deleting product. Please double check it